communication/nats/connector.go

```diff
@@ -29,14 +29,14 @@ import (
 
 // BrokerConnector establishes new connections to NATS servers and handles reconnects.
 type BrokerConnector struct {
-	registry map[uuid.UUID]Connection
+	registry map[uuid.UUID]*ConnectionWrap
```

Comment:
Could you explain why abstract `Connection` should be replaced by exact implementation?

Reply:
Because conn.Conn.Flush() call is not available on the interface.

Reply:
Yes, and it's nats.Connector, which holds exact implementations, not abstract ones